### PR TITLE
Kw pomodoro label and description

### PR DESCRIPTION
**Pull request by rodrigosiqueira**
This is part of the work to add Pomodoro with a report to users. This PR intends to:

1. Add labels to a specific timebox;
2. Add description associated to a label;
3. Store data.

@magalilemes since you're working on things related to data, this PR might be something that you want to take a deep look at.

### CODE REVIEW

File: tests/kw_string_test.sh [hunk 1]
```diff
@@ -60,4 +63,54 @@ function str_is_a_number_Test
...
+function str_length_Test
+{
+  local output
+  local string_sample="let's check the string lenght" # 29
+
+  output=$(str_length "$string_sample")
+  assert_equals_helper 'String lenght did not match' "$LINENO" '29' "$output"
+
+  output=$(str_length '')
+  assert_equals_helper 'Empty string should be 0 lenght' "$LINENO" '0' "$output"
```
Comment (magalilemes) [inline tests/kw_string_test.sh:75]: lenght should be length

File: src/pomodoro.sh [hunk 2]
```diff
@@ -228,6 +281,17 @@ function pomodoro_parser()
...
+      exit 22 # EINVAL
+    fi
+  fi
+
+  options_values['LABEL']=$(str_strip "${options_values['LABEL']}")
+  options_values['DESCRIPTION']=$(str_strip "${options_values['DESCRIPTION']}")
+
   if [[ "${options_values['TIMER']}" != 0 && "${options_values['SHOW_TIMER']}" == 1 ]]; then
     warning '--current|-c is ignored when used with --set-timer,t'
     options_values['SHOW_TIMER']=0
```
Comment (magalilemes) [inline src/pomodoro.sh:297]: `pomodoro_help()` has to be updated to show the new options

**Review by magalilemes — CHANGES_REQUESTED** [review]
Everything seems to be alright. That's an awesome patch! 🙃

File: documentation/man/kw.rst [hunk 3]
```diff
@@ -360,6 +360,16 @@ indicates the timer for the Pomodoro timebox. It is **mandatory** to add a
 suffix that indicates the time unit, which can be h (hour), m (minutes), or s
 (seconds).
 
+One helpful feature associated with timers is the *--label|-l** option since it
```
Comment (kwy95) [inline documentation/man/kw.rst:363]: Shouldn't `*--label|-l**` be `**--label|-l**`? Or `*--label|-l*`?

File: documentation/man/kw.rst [hunk 4]
```diff
@@ -349,8 +349,8 @@ displays the current month data.
 4. *--year [YEAR]*: exhibits the current year summary if the user does not
 specify a year.
 
-p, pomodoro --set-timer|-t INTEGER[h|m|s] | --current
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+p, pomodoro --set-timer|-t INTEGER[h|m|s] [--label|-l STRING [--description|-d STRING] ] | --current
```
Comment (kwy95) [inline documentation/man/kw.rst:352]: Here it isn't explicit that you can use `--current` or `-c`

**Review by kwy95 — CHANGES_REQUESTED** [review]

**Comment by rodrigosiqueira** [comment]
Hi,
I updated this entire series.
@magalilemes @joaoseckler @alan-barzilay @kwy95 could you review it?

File: src/kw_string.sh [hunk 5]
```diff
@@ -25,6 +25,46 @@ function last_char()
 # Returns 0 if the string is a number, otherwise, return 1.
 function str_is_a_number()
 {
-  [[ "$1" =~ ^[0-9]+$ ]] && return 0
+  local value="$1"
+
+  value=$(echo "$value" | xargs)
```
Comment (kwy95) [inline src/kw_string.sh:30]: Could this use the function `str_trim` ?

**Comment by alan-barzilay** [comment]
the kw file is not using include for the pomodoro flag, since this PR is fixing the use of include inside the pomodoro file, maybe we should add this change to the kw file as well @rodrigosiqueira

**Pull request merged**